File: frontend/src/types/historian.ts
export type ConfidenceLevel = "high" | "medium" | "low";
export type ClassificationType = "verified" | "inferred" | "human_knowledge";

export interface EvidenceItem {
  source_type: "commit" | "file" | "pull_request" | "issue" | "knowledge_item" | string;
  source_id: string;
  excerpt?: string | null;
}

export interface HistorianQuestionRequest {
  question: string;
}

export interface HistorianQuestionResponse {
  answer: string;
  evidence: EvidenceItem[];
  confidence: ConfidenceLevel;
  classification: ClassificationType;
  is_refusal?: boolean;
  gap_detected?: boolean;
  gap_id?: string | null;
}

File: backend/app/schemas/historian.py
"""Historian question API schemas."""

from pydantic import BaseModel, Field


class HistorianQuestionRequest(BaseModel):
    question: str = Field(..., min_length=3)


class EvidenceItem(BaseModel):
    source_type: str
    source_id: str | None = None
    excerpt: str | None = None


class HistorianQuestionResponse(BaseModel):
    answer: str
    evidence: list[EvidenceItem]
    confidence: str
    classification: str
